src/metabase/api/collection.clj

```diff
@@ -75,25 +75,35 @@
   the root, if `collection-id` is `nil`) and its immediate children, to avoid reading the entire collection tree when it
   is not necessary.
 
-  For archived, we can either pass in include both (when archived is nil), include only archived is true, or archived is false."
-  [archived exclude-other-user-collections namespace shallow collection-id]
-  (cond->>
-   (t2/select :model/Collection
-              {:where [:and
-                       (when (some? archived)
-                         [:= :archived archived])
-                       (when shallow
-                         (location-from-collection-id-clause collection-id))
-                       (when exclude-other-user-collections
-                         [:or [:= :personal_owner_id nil] [:= :personal_owner_id api/*current-user-id*]])
-                       (perms/audit-namespace-clause :namespace namespace)
-                       (collection/visible-collection-ids->honeysql-filter-clause
-                        :id
-                        (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]
-               ;; Order NULL collection types first so that audit collections are last
-               :order-by [[[[:case [:= :type nil] 0 :else 1]] :asc]
-                          [:%lower.name :asc]]})
-    exclude-other-user-collections (remove-other-users-personal-subcollections api/*current-user-id*)))
+  For `archived`, we can either include all values (when `archived` is `nil`),
+  or to include when `archived` is `true` or `false`.
+
+  To select only personal collections, pass in `personal-only` as `true`. This is disregard other parameters."
+  [{:keys [archived exclude-other-user-collections namespace shallow collection-id personal-only]}]
+  (if personal-only
+    (t2/select :model/Collection
+               {:where [:and
+                        [:!= :personal_owner_id nil]
+                        (collection/visible-collection-ids->honeysql-filter-clause
+                         :id
+                         (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]})
```

Comment:
I won't explicitly ask you to change it, but we've been trying to pass dynamic vars into functions more, in other words that would mean passing in the permissions set as an arg to `select-collections`. 

I don't see why we should ignore the other args when `personal-only`, but there's probably a good reason.

Reply:
Currently they don't make sense in terms of business logic (i.e. you can't archive a personal collection), so I think this made more sense than confusing API callers.

Reply:
Thinking about it a bit more I think it makes the logic much cleaner that maybe we allow it even if some of the combinations make no sense. I'll change it